### DISCUSSION TITLE
with_timeout Usage

### DISCUSSION THREAD

**Discussion by PiyushSaini09** [cocotb/cocotb#2825] [Q&A] [answered]
Hi-
I was trying to incorporate `with_timeout` in my design using cocotb v1.6.0.
Wrote the function as `await with_timeout(checkACK(dut),1,'ms')` whre checkACK is my coroutine. Kept a timeout interval of 1ms. But after executing, I encounter the following error:
`TypeError: All triggers must be instances of Trigger! Got: coroutine`

I dont completely understand this error. I triggered my coroutine so why is it throwing an error? 
As per the documentation, the first argument should be coroutine. 
![image](https://user-images.githubusercontent.com/96406295/148552904-20907038-f55e-4ad2-8d67-af603b5e1043.png)

What am I doing wrong?

**Comment by marlonjames** [comment, marked as answer]
Support for Python `coroutine` objects was added after the 1.6 release, so it is only on master.
It was added in https://github.com/cocotb/cocotb/pull/2494.

You'll want to wrap `checkAck(dut)` with `cocotb.start_soon()` to create a cocotb Task that can then be passed in.
